src/vaadin-overlay.html

```diff
@@ -247,6 +246,9 @@
 
       static get properties() {
         return {
+          /**
+           * When true, the overlay is visible and attached to body.
+           */
           opened: {
             type: Boolean,
```

Comment:
This currently results in:
```ts
opened: boolean|null|undefined;
```

We could add `@type {boolean}` to get:
```ts
opened: boolean;
```
though this would only make sense if it had a default value instead of being undefined by default. I'm not sure why this doesn't have a default value of `false`, but if we don't want to change that behaviour now, then I'd prefer at least `@type {boolean | undefined}` to get:
```ts
opened: boolean|undefined;
```

```suggestion
          /**
           * When true, the overlay is visible and attached to body.
           * @type {boolean | undefined}
           */
          opened: {
            type: Boolean,
```

Reply:
As long as `boolean|null|undefined` is a default output by the generator tool, I don't think we want to always make it explicit by using `boolean | undefined`.

I can update but then we'd have to remember to do it for all `boolean` properties.

Reply:
This is not so important, but I would prefer to have stricter types where possible so the API is more clear and there's less chance for confusion about why some variable can be null/undefined or what it means in those cases.

If this was new API, optimal would be `@type {boolean}` with a default value of `false`.

Here this case just happens to be a boolean, but the same thinking/logic would apply for any other type, but with non primitive types it seems you'd have to also use `!` to get rid of the default nullable behaviour.

I guess the default output of the generator is quite liberal in the allowed values for a type on purpose so that it's easier to add type defs where the output is on the safer side if you don't think too much about the null/undefined cases.

I'm ok with leaving these booleans in this PR as is though, if you disagree. You can mark this as resolved if you don't want to change this now.

Reply:
Regarding default values, I would not expect default values to end up in `d.ts` files.

Added `@type {boolean]` for properties that have `default: false` (not this one).

Reply:
> I would not expect default values to end up in d.ts files.
Me neither (except maybe for method parameters if they have a default value). I didn't mean having the default in the d.ts.